From db5b5f265aada2e11c3f318f4e8730adcec164fb Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Mon, 10 Jul 2017 18:16:23 +0200
Subject: [PATCH 39/69] kirkwood: upgrade: fix RAMFS_COPY_*

RAMFS_COPY_* are moved to platform.sh toplevel. The nand_do_upgrade call is
moved to platform_do_upgrade.

Fixes: 30f61a34b4cf "base-files: always use staged sysupgrade"
Signed-off-by: Matthias Schiffer <mschiffer@universe-factory.net>
---
 target/linux/kirkwood/base-files/lib/upgrade/linksys.sh |  8 --------
 .../linux/kirkwood/base-files/lib/upgrade/platform.sh   | 17 ++++-------------
 2 files changed, 4 insertions(+), 21 deletions(-)

diff --git a/target/linux/kirkwood/base-files/lib/upgrade/linksys.sh b/target/linux/kirkwood/base-files/lib/upgrade/linksys.sh
index 13a8f1cb55..dde3bd1b21 100644
--- a/target/linux/kirkwood/base-files/lib/upgrade/linksys.sh
+++ b/target/linux/kirkwood/base-files/lib/upgrade/linksys.sh
@@ -69,11 +69,3 @@ platform_do_upgrade_linksys() {
 		fi
 	}
 }
-
-linksys_preupgrade() {
-	export RAMFS_COPY_BIN="${RAMFS_COPY_BIN} /usr/sbin/fw_printenv /usr/sbin/fw_setenv"
-	export RAMFS_COPY_BIN="${RAMFS_COPY_BIN} /bin/mkdir /bin/touch"
-	export RAMFS_COPY_DATA="${RAMFS_COPY_DATA} /etc/fw_env.config /var/lock/fw_printenv.lock"
-}
-
-append sysupgrade_pre_upgrade linksys_preupgrade
diff --git a/target/linux/kirkwood/base-files/lib/upgrade/platform.sh b/target/linux/kirkwood/base-files/lib/upgrade/platform.sh
index e7c6a1120a..c8c66c1b9d 100644
--- a/target/linux/kirkwood/base-files/lib/upgrade/platform.sh
+++ b/target/linux/kirkwood/base-files/lib/upgrade/platform.sh
@@ -1,5 +1,8 @@
 . /lib/kirkwood.sh
 
+RAMFS_COPY_BIN='/usr/sbin/fw_printenv /usr/sbin/fw_setenv'
+RAMFS_COPY_DATA='/etc/fw_env.config /var/lock/fw_printenv.lock'
+
 platform_check_image() {
 	[ "$#" -gt 1 ] && return 1
 	local board="$(kirkwood_board_name)"
@@ -33,19 +36,7 @@ platform_do_upgrade() {
 		platform_do_upgrade_linksys "$ARGV"
 		;;
 	*)
-		default_do_upgrade "$@"
-		;;
-	esac
-}
-
-platform_pre_upgrade() {
-	local board=$(kirkwood_board_name)
-
-	case "$board" in
-	"linksys-audi"|\
-	"linksys-viper") ;;
-	*)
-		nand_do_upgrade $1
+		nand_do_upgrade "$ARGV"
 		;;
 	esac
 }
-- 
2.16.2

